src/cryptographic_primitives/commitments/hash_commitment.rs

```diff
@@ -33,20 +36,73 @@ impl Commitment for HashCommitment {
 
         BigInt::from(digest.finish().as_ref())
     }
+
+    fn create_commitment(
+        message: &BigInt) -> (BigInt, BigInt)
+    {
+        let mut digest = Context::new(&SHA256);
+        let bytes_message: Vec<u8> = message.into();
+        digest.update(&bytes_message);
+
+        let blinding_factor = &(BigInt::sample(SECURITY_BITS));
+        let bytes_blinding_factor: Vec<u8> = blinding_factor.into();
+        digest.update(&bytes_blinding_factor);
+
+        (BigInt::from(digest.finish().as_ref()), blinding_factor.clone())
+    }
 }
 
 #[cfg(test)]
 mod tests {
     use ::BigInteger as BigInt;
     use super::Commitment;
     use super::HashCommitment;
+    use arithmetic::traits::Samplable;
+    const SECURITY_BITS : usize = 256;
```

Comment:
Can you extract this at the mod.rs level. Then you can use it in both unit test and implementation